File: challenges/3-tipos-de-triangulo/ruby/mcruvinel/Desafio-3.rb
p "Informe tres valores para verificar qual o tipo de triangulo"
p "Infome a primeira medida: "
lado_a = Float(gets)
p "Infome a segunda medida: "
lado_b = Float(gets)
p "Infome a terceira medida: "
lado_c = Float(gets)

lados = [lado_a, lado_b, lado_c]
lados_ordenados = lados.sort { |a, b| b <=> a }

if lado_a >= lado_b + lado_c
    p 'NAO FORMA TRIANGULO'
  else
    p 'TRIANGULO RETANGULO' if lado_a == lado_b + lado_c
    p 'TRIANGULO OBTUSANGULO' if lado_a > lado_b + lado_c
    p 'TRIANGULO ACUTANGULO' if lado_a < lado_b + lado_c
    p 'TRIANGULO EQUILATERO' if lado_a == lado_b && lado_b == lado_c
    p 'TRIANGULO ISOSCELES' if (lado_a == lado_b && lado_b != lado_c) || (lado_a == lado_c && lado_c != lado_b) || (lado_c == lado_b && lado_b != lado_a)
end

File: challenges/3-tipos-de-triangulo/Python/Desafio3.py
#Desafio3
print("Insira as medidas para descobrir o tipo de triângulo:")

a = float(input("Primeiro lado:"))
b = float(input("Segundo lado:"))
c = float(input("Terceiro lado:"))

lado_triangulo = [a, b, c]
lado_triangulo.sort(reverse = True)

a = lado_triangulo[0]
b = lado_triangulo[1]
c = lado_triangulo[2]

if a >= b + c:
  print("NAO FORMA TRIANGULO")
else:
  if a ** 2 == b ** 2 + c ** 2:
    print("TRIANGULO RETANGULO")
  elif a ** 2 > b ** 2 + c ** 2:
    print("TRIANGULO OBTUSANGULO")
  elif a ** 2 < b ** 2 + c ** 2:
    print("TRIANGULO ACUTANGULO")